server/src/storage_engine/mod.rs

```diff
@@ -22,9 +18,25 @@ pub trait StorageEngine {
 
     fn write_file(&self, path: String, data: &[u8]) -> Result<(), EngineError>;
 
-    fn delete_file(&self, path: String) -> Result<(), EngineError>;
+    fn delete_directory_recursive(&self, path: String) -> Result<(), EngineError>;
 
-    fn delete_directory(&self, path: String) -> Result<(), EngineError>;
+    fn is_exist(&self, path: String) -> Result<bool, EngineError>;
 
-    fn delete_directory_recursive(&self, path: String) -> Result<(), EngineError>;
+    fn directory_empty(&self, path: String) -> Result<(), EngineError>;
```

Comment:
it's a good one. But it doesn't need to add to this trait.

Reply:
done